src/Microsoft.DotNet.Watcher.Tools/Program.cs

```diff
@@ -60,54 +67,58 @@ public static int Main(string[] args)
+                if (_cts.IsCancellationRequested)
+                {
+                    return 1;
+                }
+
+                if (options.ListFiles)
                 {
-                    if (!ctrlCTokenSource.IsCancellationRequested)
-                    {
-                        reporter.Output("Shutdown requested. Press Ctrl+C again to force exit.");
-                        ev.Cancel = true;
-                    }
-                    else
-                    {
-                        ev.Cancel = false;
-                    }
-                    ctrlCTokenSource.Cancel();
-                };
-
-                try
+                    return await ListFilesAsync(_reporter,
+                        options.Project,
+                        _cts.Token);
+                }
+                else
                 {
-                    if (options.ListFiles)
-                    {
-                        return await ListFilesAsync(reporter,
-                            options.Project,
-                            ctrlCTokenSource.Token);
-                    }
-                    else
-                    {
-
-                        return await MainInternalAsync(reporter,
-                            options.Project,
-                            options.RemainingArguments,
-                            ctrlCTokenSource.Token);
-                    }
+                    return await MainInternalAsync(_reporter,
+                        options.Project,
+                        options.RemainingArguments,
+                        _cts.Token);
                 }
-                catch (Exception ex)
+            }
+            catch (Exception ex)
+            {
+                if (ex is TaskCanceledException || ex is OperationCanceledException)
                 {
-                    if (ex is TaskCanceledException || ex is OperationCanceledException)
-                    {
-                        // swallow when only exception is the CTRL+C forced an exit
-                        return 0;
-                    }
-
-                    reporter.Error(ex.ToString());
-                    reporter.Error("An unexpected error occurred");
-                    return 1;
+                    // swallow when only exception is the CTRL+C forced an exit
+                    return 0;
                 }
+
+                _reporter.Error(ex.ToString());
+                _reporter.Error("An unexpected error occurred");
+                return 1;
+            }
+        }
+
+        private void OnCancelKeyPress(object sender, ConsoleCancelEventArgs args)
+        {
+            if (!_cts.IsCancellationRequested)
+            {
+                _reporter.Output("Shutdown requested. Press Ctrl+C again to force exit.");
+                // suppress default result of CTRL+C
+                args.Cancel = true;
             }
+            else
+            {
+                args.Cancel = false;
```

Comment:
nit: you could remove the `else` clause and move the `args.Cancel` assignment outside the if to `!_cts.IsCancellationRequested`.